src/middlewared/middlewared/plugins/account.py

```diff
@@ -301,6 +307,13 @@ async def query(self, filters, options):
             else:
                 entry.update({'nt_name': None, 'sid': None})
 
+            user_roles = []
+
+            for group_pk in entry['groups']:
+                user_roles.extend(group_role_info.get(group_pk, []))
```

Comment:
Is it possible that `group_pk` does not exist in `group_role_info` ? I guess it's good we are still playing it safe because this endpoint is critical but just curious